proxy/forward_auth.go

```diff
@@ -143,3 +141,10 @@ func (p *Proxy) Verify(verifyOnly bool) http.Handler {
 		return nil
 	})
 }
+
+func (p *Proxy) getOriginalRequest(r *http.Request, originalURL *url.URL) *http.Request {
```

Comment:
Could you point me in the direction of where you first saw this copy-request pattern. I need to get in the habit of this. Would like to better understand when and where it makes sense.

Reply:
It's probably not necessary... but I saw it with the reverse proxy code. https://github.com/golang/go/issues/20068